Sources/_FoundationInternals/String/String+Internals.swift

```diff
@@ -16,4 +16,31 @@ extension String {
             $0.properties.isWhitespace
         })
     }
+
+    static func _tryFromUTF16(_ input: UnsafeBufferPointer<UInt16>) -> String? {
+        withUnsafeTemporaryAllocation(of: UInt8.self, capacity: input.count * 3) { contents in
+            var ptr = contents.baseAddress!
+            var count = 0
+            let error = transcode(input.makeIterator(), from: UTF16.self, to: UTF8.self, stoppingOnError: true) { codeUnit in
+                ptr.pointee = codeUnit
+                ptr = ptr.advanced(by: 1)
+                count += 1
+            }
+
+            guard !error else {
+                return nil
+            }
+
+            return String._tryFromUTF8(UnsafeBufferPointer(rebasing: contents[..<count]))
+        }
+    }
+
+    static func _tryFromUTF16(_ input: UnsafeMutableBufferPointer<UInt16>, len: Int) -> String? {
```

Comment:
(Super super nit 😅) I believe Swift prefer `count` over `length`? I know this isn't a public API but I can already see us trying to remember whether it's count or length haha

Reply:
I went back and forth with this actually, so `count` it is!